include/seqan3/range/view/translation.hpp

```diff
@@ -70,82 +67,276 @@ enum class translation_frames
+                break;
+            case translation_frames_single::THIRD_FRAME_REVERSE:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 5]));
+                break;
+            default:
+                throw std::invalid_argument("Invalid type of frame. Choose from FIRST_FRAME_FORWARD, FIRST_FRAME_REVERSE, SECOND_FRAME_FORWARD, SECOND_FRAME_REVERSE, THIRD_FRAME_FORWARD and THIRD_FRAME_REVERSE.");
+                break;
         }
     }
+    //!\}
 
-    /*!\brief Pipe operator that enables view-typical use of pipe notation.
-    * \tparam irng_t The type of the range being processed.
-    * \param irange The range being processed as left argument of the pipe.
-    * \param translate_single_fn_v The result of the single-argument operator().
-    */
-    template <typename irng_t>
-    friend auto operator|(irng_t && irange, seqan3::detail::translate_single_fn const & translate_single_fn_v)
+    //!\brief Implicit conversion to container types.
+    template <typename container_type>
+    operator container_type()
+    //!\cond
+        requires container_concept<std::decay_t<container_type>> &&
+                 std::is_same_v<std::decay_t<value_type>, std::decay_t<ranges::range_value_type_t<container_type>>>
+    //!\endcond
     {
-        return translate_single_fn_v(std::forward<irng_t>(irange));
+        container_type ret;
+        ret.resize(size());
+        std::copy(cbegin(), cend(), ret.begin());
+        return ret;
     }
 };
 
+//!\brief Class template argument deduction for view_translate_single.
+template <typename urng_t>
+//!\cond
+    requires sized_range_concept<urng_t> &&
+             random_access_range_concept<urng_t> &&
+             nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<urng_t>>>>
+//!\endcond
+view_translate_single(urng_t &&, translation_frames_single const) -> view_translate_single<urng_t>;
+
+//!\brief Enable view-typical use with pipe operator.
+using translate_single_fn = declare_view_functor_type<view_translate_single>;
+
 } // namespace seqan3::detail
 
 namespace seqan3::view
 {
-/*!\brief A view performs single frame translation of nucleotide into aminoacid alphabet.
- * \tparam irng_t The type of the range being processed.
- * \param irange The range being processed.
- * \returns A range containing aminoacid sequence. See below for the properties of the returned range.
+/*!\brief A view that translates nucleotide into aminoacid alphabet for one of the six frames.
+ * \tparam urng_t The type of the range being processed.
+ * \param urange The range being processed.
+ * \param tf Translation frames (translation_frames_single::FIRST_FRAME_FORWARD,
```

Comment:
Shorter a little maybe?
"A value of seqan3::tanslation_frames that indicates the desired frames."